src/AzureStorage/Queues/AzureStorageQueue.cs

```diff
@@ -153,17 +169,17 @@ public class AzureStorageQueue<T> : QueueBase<T> where T : class {
             _workerErrorCount = 0;
         }
 
-        public override void StartWorking(Func<IQueueEntry<T>, CancellationToken, Task> handler, bool autoComplete = false, CancellationToken cancellationToken = new CancellationToken()) {
+        protected override void StartWorkingImpl(Func<IQueueEntry<T>, CancellationToken, Task> handler, bool autoComplete = false, CancellationToken cancellationToken = new CancellationToken()) {
             if (handler == null)
                 throw new ArgumentNullException(nameof(handler));
-
+            
             var linkedCancellationToken = CancellationTokenSource.CreateLinkedTokenSource(_queueDisposedCancellationTokenSource.Token, cancellationToken).Token;
 
             Task.Run(async () => {
```

Comment:
I wonder if we can make the StartWorkingImp return a task and implify the func that is passed in? I'm sure we did the func that way because it made stuff easier and simplified it a bit.. then the StartWorking can just do taskrun and call the StartWorkingImp.